### PR TITLE
CaaS API Implmentation and Command Rewrite

### PR DESCRIPTION
**Pull request by Quesos-Nachos**
Inzanit wrote the CaaS API. Cat command is pulling cats and cat gifs from Imgur now

### CODE REVIEW

File: Modix/ModixBot.cs [hunk 1]
```diff
@@ -137,6 +138,7 @@ public void LoadConfig()
             _map.AddSingleton(_config);
             _map.AddScoped<IQuoteService, QuoteService>();
             _map.AddSingleton<CodePasteService>();
+            _map.AddSingleton<IHttpClient, HttpClient>();
```
Comment (Cisien) [inline Modix/ModixBot.cs]: I'm pretty sure this usage is just asking for problems. Things like BaseUrl or DefaultRequestHeaders would be shared between different services, which could cause random weird breaks.

Should probably encapsulate that in the cat service, and inject the cat service instead.

File: appveyor.yml [hunk 2]
```diff
@@ -50,4 +50,5 @@ environment:
   log_webhook_id: 382609729802862592
   SentryToken:
     secure: TSVcrUaf7lIi4YoOKrZhqrf6MvXh+XFUMO454dwB0q8ieNFOap49KhCYhw4x9sRflgQnDhppjzdbOrFiDnwQvID4rb3nXCxnJRzPLOUwOYlM4SfitMt2LnyGzyOdminq
-
+  ImgurToken:
+    secure: 4a98fd6d74e936d079ae8cad473386ca5818283c
```
Comment (Cisien) [inline appveyor.yml]: shouldn't this be `secure: PP1+bFXhjlGEbEDdKZTIgY/cwh2cZmT1HmKoSb9e4EhcM8JAbbEfQPICuY7b/tC4`?

Comment (Cisien) [reply appveyor.yml]: Also, you need to add -e "ImgurToken=$env;ImgurToken" to the docker run in the before-deploy.ps1 script

Comment (Quesos-Nachos) [reply appveyor.yml]: I set up the initial token incorrectly. I created a second token with Postman (which hasn't been added yet). With the Imgur API, only the ClientID is needed for the authorization header. I think the client secret is only needed when POSTing stuff to Imgur

Comment (Quesos-Nachos) [reply appveyor.yml]: That might not need to be there

**Pull request merged**